### PR TITLE
WIP: Add Stats structs

### PR DESCRIPTION
**Pull request by maxhawkins**
Add structs from [rtc-domstats](https://www.w3.org/TR/webrtc/#dom-rtcstats) spec so we can begin implementing the GetStats API.

Related to #610

### CODE REVIEW

**Comment by maxhawkins** [comment]
@Sean-Der Do you think it would be useful or confusing to merge these struct definitions?

I don't have time to implement the stats just yet, but this will make it easier for someone to do it in the future. Alternately I could hang onto this PR until I have time to add a stat.

**Comment by Sean-Der** [comment]
@maxhawkins 

IMO I would merge them! someone might get interested and finish the work.

Would you mind adding some dumb tests that just assert things exist? just so coverage doesn't drop

**Comment by maxhawkins** [comment]
Will do

**Comment by maxhawkins** [comment]
PTAL

File: stats.go [hunk 1]
```diff
@@ -0,0 +1,1306 @@
...
+func (s CodecStats) ID() string {
+	return s.StatsBase.ID
+}
+
+// ReceivedRTPStreamStats contains stats measured at the receiving end of an RTP stream,
+// known either because they're measured locally or transmitted via an RTCP Receiver
+// Report (RR) or Extended Report (XR).
+type ReceivedRTPStreamStats struct {
+	// PacketsReceived is the total number of RTP packets received for this SSRC.
+	PacketsReceived uint32 `json:"packetsReceived"`
```
Comment (Sean-Der) [inline stats.go]: NIT -- What do you think of putting a newline after every member? Visually I break chunks apart by newline, so easier to consume when reading

File: stats.go [hunk 2]
```diff
@@ -0,0 +1,1306 @@
...
+	millis := float64(s)
+	nanos := int64(millis * float64(time.Millisecond))
+
+	return time.Unix(0, nanos).UTC()
+}
+
+// StatsReport collects Stats objects indexed by their ID.
+type StatsReport map[string]Stats
+
+// StatsBase contains all the fields Stats objects share
```
Comment (Sean-Der) [inline stats.go]: Does this need to be public?

File: stats.go [hunk 3]
```diff
@@ -0,0 +1,1306 @@
+package webrtc
+
+import (
+	"time"
+)
+
+// A Stats object contains a set of statistics copies out of a monitored component
+// of the WebRTC stack at a specific time.
+type Stats interface {
```
Comment (Sean-Der) [inline stats.go]: Does this need to be public?

Comment (Sean-Der) [reply stats.go]: Not a big deal, but we are putting some implementation details in the users face by having these public.

I don't feel strongly, but would be nice if `godoc` doesn't show anything we did to make our code shorter

Comment (maxhawkins) [reply stats.go]: I do want to keep the type around. Maybe i'll get rid of the methods:

```go
type Stats interface{}
```

I'm not sure Type() and Timestamp() are so useful anyway. Everyone will end up casting to the base type and they can find those values there.

The initial idea for this had to do with keeping the results of the not-yet-implemented `GetStats` method typesafe. It returns a map with polymorphic values `map[string]Stats`.

File: stats.go (hunk 2 above)
Comment (maxhawkins) [reply stats.go]: On second thought, let's just delete it.

If I get rid of the `Stats` methods like I mention below all of these fields can just be duplicated in the base stat objects.

File: stats.go (hunk 1 above)
Comment (maxhawkins) [reply stats.go]: will do

**Comment by jinleileiking** [comment]
Is this done？exciting

**Pull request merged**